x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2397 @@
+// so we need to search for the existence of the isCustomized subfield instead, which is required
+export const KQL_FILTER_PREBUILT_RULES = `${PARAMS_PREBUILT_IS_CUSTOMIZED_FIELD}: *`;
+export const KQL_FILTER_CUSTOM_RULES = `NOT ${PARAMS_PREBUILT_IS_CUSTOMIZED_FIELD}: *`;
+export const KQL_FILTER_IMMUTABLE_RULES = `${PARAMS_IMMUTABLE_FIELD}: true`;
+export const KQL_FILTER_MUTABLE_RULES = `${PARAMS_IMMUTABLE_FIELD}: false`;
+
+interface RulesFilterOptions {
+  filter: string;
+  showCustomRules: boolean;
+  showElasticRules: boolean;
+  ...
+}
+
+export function convertRulesFilterToKQL({
+  filter: searchTerm,
+  showCustomRules,
+  showElasticRules,
+  ...
+}: Partial<RulesFilterOptions>): string {
+  const kql: string[] = [];
+
+  // [... file continues ...]
+
+  if (showCustomRules && showElasticRules) {
+    // if both showCustomRules && showElasticRules selected we omit filter, as it includes all existing rules
+  } else if (showElasticRules) {
+    kql.push(`(${KQL_FILTER_PREBUILT_RULES} OR ${KQL_FILTER_IMMUTABLE_RULES})`);
+  } else if (showCustomRules) {
+    kql.push(`(${KQL_FILTER_CUSTOM_RULES} OR ${KQL_FILTER_MUTABLE_RULES})`);
+  }
+
+  // [... file continues below...]
+```
+
+Ntocie that in order to retrieve rules in which the `alert.attributes.params.prebuilt` field exists, we must rely instead on the existence of `alert.attributes.params.prebuilt.isCustomized`. This is so because the `prebuilt` field lives inside the unmapped `params` field, and KQL syntax does not allow searching for the existence of an object like `prebuilt` with the recommended syntax `alert.attributes.params.prebuilt: *`.
+
+But we can use the `prebuilt` field's `isCustomized` subfield instead, which will always exist within the object (it is required) and replace the KQL filter for `alert.attributes.params.prebuilt.isCustomized: *`.
+
+### Rule Management endpoints
+
+- **Create Rules** - `POST /rules` and **Bulk Create Rules** - `POST /rules/_bulk_create`:
+
+Currently, we don't support the `immutable` field in any of the endpoints' request parameters (except for the Import endpoint). We shouldn't support the `prebuilt` field either, because this value should be controlled by the app on the server side, and not by users.
+
+This is so because we will never want users to be able to create their own prebuilt rules, only install them, import them, and customize them. Also, a prebuilt rule should always be able to be compared to a `security-rule` asset distributed by Fleet, and receive updates from it, which would not be possible if a user creates its own prebuilt rules.
+
+Specifically, these two endpoints should be able to create only custom rules which means their `params` will look like:
+
+```ts
+// [PSEUDOCODE]
+{
+  immutable: false,
+  // prebuilt should not be in the created rule object,
+  // since this endpoint should only create custom rules
+}
+```
```

Comment:
Why users should even bother specifying any of that? The `immutable` and `prebuilt` fields should not be in the request schemas of the endpoints mentioned above:

**Create Rules** - `POST /rules` and **Bulk Create Rules** - `POST /rules/_bulk_create`

Reply:
Add a clarifiction that what is meant is the rule's SO parameters and not the request parameters of the endpoints